src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -87,8 +84,9 @@ private SafeDirectoryHandle CreateDirectoryHandle(string path)
 
         private void CloseDirectoryHandle()
         {
-            _directoryHandle?.Dispose();
-            _directoryHandle = null;
+            IntPtr handle = Interlocked.Exchange(ref _directoryHandle, IntPtr.Zero);
+            if (handle != IntPtr.Zero)
+                Interop.Sys.CloseDir(_directoryHandle);
```

Comment:
This is broken. It should be handle, not _directoryHandle.

Reply:
Likely the cause of the seg faults and test failures, as no directories are getting closed and file descriptors are being exhausted.

Reply:
Ugh, copy paste fail between VMs. Thanks!